cmake/checks/check_01_compiler_features.cmake

```diff
@@ -364,6 +364,8 @@ CHECK_CXX_SOURCE_COMPILES(
   _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wextra\\\\\\\"\")
   _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wunknown-pragmas\\\\\\\"\")
   _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wpragmas\\\\\\\"\")
+  _Pragma(\"clang diagnostic ignored \\\\\\\"-Wunknown-warning-option\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wunknown-warning\\\\\\\"\")
```

Comment:
Just too funny every time with all of those backslashes!

Reply:
But more seriously, you do `GCC diagnostic push/pop` at the top/bottom, but now you add a `clang diagnostic ignored...` in the middle. Isn't this inconsistent?

Reply:
The `clang diagnostic` option is intended for Clang alone, i.e. GCC does not have a `-Wunknown-warning-option` flag. All the other diagnostics are intended for GCC and Clang.

If I would use `GCC diagnostic` in the line with `clang diagnostic` and GCC happens to warn a bit more aggressively in a future version, we would get a similar problem.

Reply:
Are you saying that GCC is simply going to ignore the `clang` line? And that both GCC and clang understand the `GCC` lines?